omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ObsController1_9Test.java

```diff
@@ -85,6 +90,36 @@ public void getObs_shouldCreateAnObsWithACodedValueSetAsAMap() throws Exception
 		assertEquals(++originalCount, getAllCount());
 	}
 
+	@Test
+	public void shouldUpdateObs() throws Exception {
+		Drug drug = Context.getConceptService().getDrugByUuid("3cfcf118-931c-46f7-8ff6-7b876f0d4202");
+		Concept concept = Context.getConceptService().getConceptByUuid("89ca642a-dab6-4f20-b712-e12ca4fc6d36");
+		Person person = Context.getPersonService().getPerson(2);
+		final Date newDatetime = new Date();
+		Obs obs = new Obs();
+		obs.setConcept(concept);
+		obs.setValueDrug(drug);
+		obs.setValueCoded(drug.getConcept());
+		obs.setObsDatetime(newDatetime);
+		obs.setPerson(person);
+		obs.setComment("An option free text comment about the observation");
+		ObsService obsService = Context.getObsService();
+		obsService.saveObs(obs, "some random reason");
+
+		String json = "{\"concept\":\"89ca642a-dab6-4f20-b712-e12ca4fc6d36\", " +
+				"\"person\":\"5946f880-b197-400b-9caa-a3c661d23041\","
+				+ "\"obsDatetime\":\""+ DATE_FORMAT.format (newDatetime)+ "\","
+				+ "\"comment\":\"An option free text comment about the observation\"}";
+		handle(newPostRequest(getURI() + "/" + obs.getUuid(), json));
+		Obs updatedObs = obsService.getObsByUuid(obs.getUuid());
+
+		assertNotNull(updatedObs);
+		assertEquals("89ca642a-dab6-4f20-b712-e12ca4fc6d36", updatedObs.getConcept().getUuid());
```

Comment:
What have you updated? And from what value to which one?

Reply:
Oh sorry let me improve that